core/src/test/scala/org/apache/spark/storage/DiskBlockObjectWriterSuite.scala

```diff
@@ -184,4 +184,22 @@ class DiskBlockObjectWriterSuite extends SparkFunSuite with BeforeAndAfterEach {
     writer.close()
     assert(segment.length === 0)
   }
+
+  test("calling closeAndDelete() on a partial write file") {
+    val (writer, file, writeMetrics) = createWriter()
+
+    writer.write(Long.box(20), Long.box(30))
```

Comment:
Why the `Long.box`? In the previous test this was not needed. It was just a simple for like:
https://github.com/apache/spark/blob/a1e94b56e9675514ad2546fdcac8a5af36aa2b62/core/src/test/scala/org/apache/spark/storage/DiskBlockObjectWriterSuite.scala#L168-L170

What about having a for loop here and after the commit too (but with different number of records, like 100 and 50)?

Reply:
8fefa03 fix this